backend/tracim_backend/config.py

```diff
@@ -414,6 +415,21 @@ def _load_global_config(self) -> None:
             cast_func=AuthType,
             do_strip=True,
         )
+        self.USER__READ_ONLY_FIELDS: typing.Dict[str, typing.List[str]] = {}
+        for auth_type in self.AUTH_TYPES:
+            read_only = string_to_unique_item_list(
+                self.get_raw_config(f"user.profile.read_only_fields.{auth_type.value}"),
+                separator=",",
+                cast_func=str,
+                do_strip=True,
+            )
+            # HACK - M.L - 2023-10-30 - This is to satisfy CFG._check_consistency despite being
+            #  poorly usable
+            self.__setattr__(
+                f"USER__PROFILE__READ_ONLY_FIELDS__{auth_type.value.upper()}", read_only
+            )
+            self.USER__READ_ONLY_FIELDS[auth_type.value] = read_only
```

Comment:
I suggest to put an assert here in order to be sure there will be a known value. Otherwise, the dynamic behavior will make it very difficult to investigate in case of error.

Reply:
The dict parameter is to ease the usage of the said parameter. Since the use we make of it is typical to a `Dict` (Basically we are accessing a value depending on a string).

We could make it a `List` of `List`s, since `List` are used in the parameters, but I think that this is not optimal.

Reply:
We miss-understood each other.

The ease of use would be to have 4 statick config fields:

```
USER__READ_ONLY_FIELDS__DEFAULT
USER__READ_ONLY_FIELDS__LDAP
USER__READ_ONLY_FIELDS__REMOTE
USER__READ_ONLY_FIELDS__SAML
```

You feel that a dynamic configuration is better. You type the config field like this: `self.USER__READ_ONLY_FIELDS: typing.Dict[str, typing.List[str]] = {}` which does not really help because *string are evil*.

First, the typing should be something like `self.USER__READ_ONLY_FIELDS: typing.Dict[AuthType, typing.List[str]] = {}`

Then, what are the valid values for the strings? There are really no limit? 

I suggest to explicitly assert that identified read-only fields exist (raise an error during config avoid tracim to start and make it easier to be sure the configuration is ok for run)


By the way: auth types are explicitly defined in the `AuthType` enum class;  did you update `class AuthType()` with a new ENUM value SAML? (in `auth.py`)

Reply:
Ok I understand better. The reason I was not limiting anything was that I wanted to make it as evolutive as possible (as it is, this part doesn't have to be edited if anything changes). But I understand how it also *is* an issue. I will fix this

Reply:
And yes I added an auth type properly as defined in the `AuthType` enum and the `AuthType` class